### PR TITLE
Buffer Pooling Optimization

### PR DESCRIPTION
**Pull request by emac07**
Ensure that the get_next_buffer function cycles through the pool and selects the next available buffer for use. Right now, it seems to select the first available buffer without considering if both buffers are busy.

### CODE REVIEW

File: client/pool-buffer.c [hunk 1]
```diff
@@ -103,30 +103,29 @@ void destroy_buffer(struct pool_buffer *buffer) {
...
-
-	for (size_t i = 0; i < 2; ++i) {
-		if (pool[i].busy) {
-			continue;
-		}
-		buffer = &pool[i];
-	}
-
-	if (!buffer) {
-		return NULL;
```
Comment (kennylevinsen) [inline client/pool-buffer.c:117]: This return should already handle when both buffers are busy, as buffer then remains NULL.

The only change I can see is when both buffers are available, and the second buffer matches the size while the first does not. Is this the scenario you intended to optimize?

Comment (emac07) [reply client/pool-buffer.c:117]: Thank you for your review! The code snippet isn't specifically tailored to handle scenarios where both buffers are busy. It primarily addresses the condition where buffer remains NULL if unassigned. The optimization aims to improve buffer allocation efficiency, especially focusing on scenarios where both buffers are available but only one matches the required size. If that makes sense.

Comment (emac07) [reply client/pool-buffer.c:117]: I misread your message originally. Yes that is the intended scenario
.

File: client/pool-buffer.c [hunk 2]
```diff
@@ -103,30 +103,29 @@ void destroy_buffer(struct pool_buffer *buffer) {
...
+        if (!pool[i].busy &&
+            (pool[i].width == width && pool[i].height == height)) {
+            pool[i].busy = true;
+            return &pool[i];
+        }
+    }
+
+    for (size_t i = 0; i < 2; ++i) {
+        if (!pool[i].busy) {
+            if (pool[i].width != width || pool[i].height != height) {
```
Comment (kennylevinsen) [inline client/pool-buffer.c:118]: This condition is always true, as we would have returned from the function during the first loop if a non-busy buffer had matching dimensions.

**Pull request closed without merging**